src/qibotn/backends/cpu.py

```diff
@@ -0,0 +1,144 @@
+
+    def execute_circuit(
+        self, circuit, initial_state=None, nshots=None, return_array=False
+    ):  # pragma: no cover
+        """Executes a quantum circuit.
+
+        Args:
+            circuit (:class:`qibo.models.circuit.Circuit`): Circuit to execute.
+            initial_state (:class:`qibo.models.circuit.Circuit`): Circuit to prepare the initial state.
+                If ``None`` the default ``|00...0>`` state is used.
+
+        Returns:
+            xxx.
+        """
+
+        import qibotn.eval_qu as eval
+
+        if (
+            self.MPI_enabled == False
+            and self.MPS_enabled == False
+            and self.NCCL_enabled == False
+            and self.expectation_enabled == False
+        ):
+
+            state = eval.dense_vector_tn_qu(
+                circuit.to_qasm(), initial_state, is_mps=False, backend="numpy"
+            )
+
+        elif (
+            self.MPI_enabled == False
+            and self.MPS_enabled == True
+            and self.NCCL_enabled == False
+            and self.expectation_enabled == False
+        ):
+
+            state = eval.dense_vector_tn_qu(
+                circuit.to_qasm(), initial_state, is_mps=True, backend="numpy"
+            )
+
+        elif (
+            self.MPI_enabled == True
+            and self.MPS_enabled == False
+            and self.NCCL_enabled == False
+            and self.expectation_enabled == False
+        ):
+
+            raise_error(NotImplementedError, "QiboTN quimb backend cannot support MPI.")
+
+        elif (
+            self.MPI_enabled == False
+            and self.MPS_enabled == False
+            and self.NCCL_enabled == True
+            and self.expectation_enabled == False
+        ):
+
+            raise_error(
+                NotImplementedError, "QiboTN quimb backend cannot support NCCL."
+            )
+
+        elif (
+            self.MPI_enabled == False
+            and self.MPS_enabled == False
+            and self.NCCL_enabled == False
+            and self.expectation_enabled == True
+        ):
+
+            raise_error(
+                NotImplementedError, "QiboTN quimb backend cannot support expectation"
+            )
+
+        elif (
+            self.MPI_enabled == True
+            and self.MPS_enabled == False
+            and self.NCCL_enabled == False
+            and self.expectation_enabled == True
+        ):
+            raise_error(
+                NotImplementedError, "QiboTN quimb backend cannot support expectation"
+            )
+
+        elif (
+            self.MPI_enabled == False
+            and self.MPS_enabled == False
+            and self.NCCL_enabled == True
+            and self.expectation_enabled == True
+        ):
+            raise_error(
+                NotImplementedError, "QiboTN quimb backend cannot support expectation"
+            )
+        else:
+            raise_error(NotImplementedError, "Compute type not supported.")
```

Comment:
Essentially, the only feature supported is `MPS_enabled`.

I'd make this whole switch simpler, by avoiding exploring each case on its own:
```suggestion
        if self.MPI_enabled == True:
            raise_error(NotImplementedError, "QiboTN quimb backend cannot support MPI.")
        if self.NCCL_enabled == True:
            raise_error(
                NotImplementedError, "QiboTN quimb backend cannot support NCCL."
            )
        if self.expectation_enabled == True:
            raise_error(
                NotImplementedError, "QiboTN quimb backend cannot support expectation"
            )

       state = eval.dense_vector_tn_qu(
                circuit.to_qasm(), initial_state, is_mps=self.MPS_enabled, backend="numpy"
       )
```